R/teal_data-get_code.R

```diff
@@ -1,12 +1,12 @@
 #' Get code from `teal_data` object
 #'
-#' Retrieve code from `teal_data` object.
+#' `r lifecycle::badge("deprecated")` Use [`teal.code::get_code()`]. Retrieve code from `teal_data` object.
```

Comment:
hold on here
are we deprecating the whole method?

Reply:
Well, this method had one extra parameter. Once this parameter is deprecated, we actually use `teal.code::get_code()` and we introduce this parameter in there. So the whole `teal.data::get_code` will be deprecated.

Reply:
I think the logic should be adjusted, so that it warns on the whole usage of the function
```r
      lifecycle::deprecate_warn(
        when = "0.6.1",
        what = "teal.data::get_code()",
        with = "teal.code::get_code()",
        always = TRUE
      )
```

Reply:
and I think we can move out `lifecycle::deprecate_warn` outside of the `lifecycle::is_present(datanames)` so it will be as follows

```r
setMethod("get_code",
  signature = "teal_data",
  definition = function(object, deparse = TRUE, names = NULL, datanames = lifecycle::deprecated(), ...) {
    lifecycle::deprecate_warn(
      when = "0.6.1",
      what = "teal.data::get_code()",
      with = "teal.code::get_code()",
      always = TRUE
    )
    if (lifecycle::is_present(datanames)) {
      names <- datanames
    }

    checkmate::assert_character(names, min.len = 1L, null.ok = TRUE)
    checkmate::assert_flag(deparse)

    callNextMethod(object = object, deparse = deparse, names = names, ...)
  }
)
```

Reply:
That actually makes sense to me! However, I might be too far away from the code base to spot any potential issues. Maybe other devs could help on that?

Reply:
What's your take on this @gogonzo ?

Reply:
This will throw a warning when `get_code` is used without prefix.

```r
pkgload::load_all("teal.code")
pkgload::load_all("teal.data")

data <- teal_data() |> within(a <- 1)
data |> get_code()
# Warning message:
# `get_code()` was deprecated in teal.data 0.6.1.
```

We don't want to throw this warning whenever `get_code()` is run on the `teal_data` object. Probably you want to throw only when `teal.data::get_code()` is used, but then you need to do `sys.call` or something to detect this - I don't recommend, I'd prefer to just deprecate `datanames` and someday this function will just disappear from the package.

Reply:
Ok @gogonzo I though we want to deprecate the whole function, as it doesn't have any specific purpose right now.
So you see the warning every time you use `get_code` so that you know you should move to teal.code.

Reply:
Ok, I'm having second thoughts. 
Let's just deprecate the parameter right now, as it's safer. And then let's remove the function later, someday

Reply:
Just changed the documentation and the implementation, so only the usage of the parameter is deprecated.
https://github.com/insightsengineering/teal.data/pull/343/commits/5aa5a7f06dd4877c9a03fc260c4e5101ac2a4a1d